### PR TITLE
add ClientRequestBuilder::form()

### PR DESCRIPTION
**Pull request by ghost**
#276

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -586,6 +590,47 @@ impl From<JsonError> for JsonPayloadError {
     }
 }
 
+/// A set of errors that can occur during parsing json payloads
+#[derive(Fail, Debug)]
+pub enum FormPayloadError {
```
Comment (fafhrd91) [inline src/error.rs]: where do you use this error?

Comment (ghost) [reply src/error.rs]: oh sorry. I thought it was necessary because it was implemented for JsonError. I am deleting it.

Comment (fafhrd91) [reply src/error.rs]: Form extractor uses `UrlencodedError` error

**Comment by fafhrd91** [comment]
looks good!  could you also add test case and update changes.md

**Comment by ghost** [comment]
Ok. I'm working on it.

**Comment by ghost** [comment]
I can't find a example for json() method. I don't know how to write one for this one.

**Comment by fafhrd91** [comment]
I think you should write integration test, something like https://github.com/actix/actix-web/blob/master/tests/test_handlers.rs#L73

**Comment by fafhrd91** [comment]
this line construct ClientRequest https://github.com/actix/actix-web/blob/master/tests/test_handlers.rs#L85

replace body with form

**Comment by ghost** [comment]
I have to do something for 1 hour after that I can do this.

**Review by fafhrd91 — APPROVED** [review]

**Comment by fafhrd91** [comment]
thanks!

**Comment by ghost** [comment]
Sorry for unnecessary commits. cargo check didn't show these errors. My computer is really bad so I cant run cargo test.

**Comment by fafhrd91** [comment]
You can use `cargo check —tests`

**Comment by ghost** [comment]
thanks. this one should pass.

**Comment by fafhrd91** [comment]
thanks!

**Pull request merged**